pkg/controller/master/node/promote_controller.go

```diff
@@ -68,18 +71,21 @@ type PromoteHandler struct {
 	jobCache  ctlbatchv1.JobCache
 	recorder  record.EventRecorder
 	namespace string
+	apps      catalogv1.AppController
```

Comment:
Since we don't need to update `App` in this controller, could we just use `catalogv1.AppCache`? Thanks.